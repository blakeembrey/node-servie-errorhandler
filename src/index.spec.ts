import { Request } from "servie/dist/node";
import * as boom from "boom";
import * as httpErrors from "http-errors";
import { errorhandler } from "./index";

describe("servie-errorhandler", () => {
  const req = new Request("/");
  const basicHandler = errorhandler(req);

  it('should fail "gracefully" with non-error', () => {
    const res = basicHandler("test");

    expect(res).toMatchSnapshot();
  });

  it("should render an error", () => {
    const res = basicHandler(new Error("boom!"));

    expect(res).toMatchSnapshot();
  });

  it("should render boom status errors", () => {
    const res = basicHandler(boom.badRequest("data has an issue"));

    expect(res).toMatchSnapshot();
  });

  it("should render http errors status error", () => {
    const res = basicHandler(new httpErrors.BadRequest("data has an issue"));

    expect(res).toMatchSnapshot();
  });
});
