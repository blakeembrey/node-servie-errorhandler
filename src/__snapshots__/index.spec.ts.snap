// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servie-errorhandler should fail "gracefully" with non-error 1`] = `
Response {
  "$rawBody": "<!doctype html>
<html lang=\\"en\\">
<head><meta charset=\\"utf-8\\"><title>Error</title></head>
<body>
<pre>{
 &nbsp;&quot;status&quot;: 500,
 &nbsp;&quot;error&quot;: &quot;Internal Server Error&quot;,
 &nbsp;&quot;message&quot;: &quot;Error&quot;
}</pre>
</body>
</html>",
  "headers": Headers {
    "object": Object {
      "content-length": "265",
      "content-security-policy": "default-src 'self'",
      "content-type": "text/html",
      "x-content-type-options": "nosniff",
    },
  },
  "status": 500,
  "statusText": "",
  "trailer": Promise {},
}
`;

exports[`servie-errorhandler should render an error 1`] = `
Response {
  "$rawBody": "<!doctype html>
<html lang=\\"en\\">
<head><meta charset=\\"utf-8\\"><title>Error</title></head>
<body>
<pre>{
 &nbsp;&quot;status&quot;: 500,
 &nbsp;&quot;error&quot;: &quot;Internal Server Error&quot;,
 &nbsp;&quot;message&quot;: &quot;boom!&quot;
}</pre>
</body>
</html>",
  "headers": Headers {
    "object": Object {
      "content-length": "265",
      "content-security-policy": "default-src 'self'",
      "content-type": "text/html",
      "x-content-type-options": "nosniff",
    },
  },
  "status": 500,
  "statusText": "",
  "trailer": Promise {},
}
`;

exports[`servie-errorhandler should render boom status errors 1`] = `
Response {
  "$rawBody": "<!doctype html>
<html lang=\\"en\\">
<head><meta charset=\\"utf-8\\"><title>Error</title></head>
<body>
<pre>{
 &nbsp;&quot;statusCode&quot;: 400,
 &nbsp;&quot;error&quot;: &quot;Bad Request&quot;,
 &nbsp;&quot;message&quot;: &quot;data has an issue&quot;
}</pre>
</body>
</html>",
  "headers": Headers {
    "object": Object {
      "content-length": "271",
      "content-security-policy": "default-src 'self'",
      "content-type": "text/html",
      "x-content-type-options": "nosniff",
    },
  },
  "status": 400,
  "statusText": "",
  "trailer": Promise {},
}
`;

exports[`servie-errorhandler should render http errors status error 1`] = `
Response {
  "$rawBody": "<!doctype html>
<html lang=\\"en\\">
<head><meta charset=\\"utf-8\\"><title>Error</title></head>
<body>
<pre>{
 &nbsp;&quot;status&quot;: 400,
 &nbsp;&quot;error&quot;: &quot;Bad Request&quot;,
 &nbsp;&quot;message&quot;: &quot;data has an issue&quot;
}</pre>
</body>
</html>",
  "headers": Headers {
    "object": Object {
      "content-length": "267",
      "content-security-policy": "default-src 'self'",
      "content-type": "text/html",
      "x-content-type-options": "nosniff",
    },
  },
  "status": 400,
  "statusText": "",
  "trailer": Promise {},
}
`;
